### PR TITLE
Auth Page

### PR DESCRIPTION
**Pull request by dubem7**
To view the changes go to http://localhost:3000/#/

There is some code to be removed after review but the Modals would have otherwise been inaccessible.

### CODE REVIEW

File: src/components/Auth/Auth.module.scss
Comment (radamson) [inline src/components/Auth/Auth.module.scss:3]: This should be a `styles.tsx` if I'm not mistaken.  We've switched from using scss to makestyles/jss.

Comment (blangley28) [reply src/components/Auth/Auth.module.scss:3]: Once you do that also update the colors use variables. There are a bunch already defined in the theme and if you want to add more they can be added there

File: src/components/Auth/AuthBody.tsx
Comment (blangley28) [inline src/components/Auth/AuthBody.tsx:25]: There are fontawesome icons for these if we want to use them instead. I don't know if there's any difference

File: src/components/Auth/AuthBody.tsx
Comment (blangley28) [inline src/components/Auth/AuthBody.tsx:17]: This should be a comma
```suggestion
      <label className={styles.subText}>using mCODE, a standardized data </label>
```

File: src/components/Auth/AuthFooter.tsx
Comment (blangley28) [inline src/components/Auth/AuthFooter.tsx:15]: Using a call back here (and in the `Link` below) instead of an arrow function will improve re-rendering the Link component

File: src/components/Auth/AuthHeader.tsx
Comment (blangley28) [inline src/components/Auth/AuthHeader.tsx:18]: Instead of defining like 6 different modals can it not just be one modal which renders differently based on the props?

File: src/index.js
Comment (blangley28) [inline src/index.js:25]: Personally I think the landing page should be `/`. That way when you just go directly to the base url you are prompted with the splash screen

File: src/components/Auth/styles.tsx
Comment (blangley28) [inline src/components/Auth/styles.tsx:1]: Is this replacing the scss file? Once its been fully moved over be sure to delete that file

**Review by blangley28 — COMMENTED** [review]
I know its a WIP so perhaps you already know this but I get the following error when I try and load http://localhost:3000/#/login

Screen Shot 2020-10-27 at 11 53 07 AM

File: src/components/Auth/Landing.tsx
Comment (blangley28) [inline src/components/Auth/Landing.tsx:64]: There are a bunch of wdyr issues I think they're all related to using arrow functions in the props instead of using callbacks

File: src/components/Auth/LandingFooter.tsx
Comment (blangley28) [inline src/components/Auth/LandingFooter.tsx:14]: Clicking on either of these links gives 
`TypeError: resetCurrentCriteria is not a function`
and then does not load the website in the new tab. I think its because the routing happens in `index.js` so the App is never loaded (which is where the currentCriteriaContext is stored).

In fact, why does this component even need to resetCurrentCriteria()?

File: src/components/Auth/NewPasswordModal.tsx
Comment (blangley28) [inline src/components/Auth/NewPasswordModal.tsx:5]: Use absolute imports here (and in all of the modals)
```suggestion
import Modal from 'components/elements/Modal';
import TextInput from 'components/elements/TextInput';
```

**Review by blangley28 — CHANGES_REQUESTED** [review]
Since this also implements the /demo endpoint we will need to sync it with #64 which also implements the /demo endpoint but only for components in `components/StaticApp`

**Comment by noranda** [comment]
Just a note that this should be good to review now. You should now go to `/` for the landing page and `/demo` for the current demo. You can also switch back and forth between them by clicking on the logos in the header.

File: src/components/Auth/LandingBody.tsx
Comment (blangley28) [inline src/components/Auth/LandingBody.tsx:34]: Can we have this redirect to `/demo`. I know its technically part of the other task but just so we have a way to get to the actual builder.

Comment (blangley28) [reply src/components/Auth/LandingBody.tsx:34]: I know see your comment above, I guess there is a way so its up to you whether to implement this here or just let PATHWAYS-389 implement it

Comment (noranda) [reply src/components/Auth/LandingBody.tsx]: I think it's a good idea, so I implemented it. :)

File: src/components/Auth/NewPasswordModal.tsx
Comment (noranda) [reply src/components/Auth/NewPasswordModal.tsx]: Oops, fixed all these imports.

File: src/components/Auth/LandingFooter.tsx
Comment (noranda) [reply src/components/Auth/LandingFooter.tsx:14]: Oops again, fixed!

File: src/components/Auth/LandingFooter.tsx
Comment (blangley28) [inline src/components/Auth/LandingFooter.tsx:15]: This still hangs for me. I think it needs to be 
```suggestion
        href="https://www.mitre.org/"
```

**Review by blangley28 — APPROVED** [review]

**Comment by dehall** [comment]
This all looks good. The only thing I'm thinking is how will the routing structure work once we have user auth in place? I assume if someone is already logged in and goes to the root URL, they will either see their pathways/criteria there, or get redirected to the pathways/criteria list. I'm wondering if there's anything simple we could do here in preparation for that, and potentially avoid some future rework.

**Comment by dehall** [comment]
2 more specific questions:
1 - Are the TODOs in this meant to be changed before merging this, or once the backend auth process is in place?
2 - Can we disable the login/signup buttons until the process is ready? Or possibly make it configurable to hide them? I don't expect we're going to have a public-facing backend server anytime soon, so the login/sign up buttons will never work on the github pages demo

**Comment by dubem7** [comment]
> 1 - Are the TODOs in this meant to be changed before merging this, or once the backend auth process is in place? 


I was originally thinking before we merge. I just wanted those modals to be accessible while it was up for review.

> 2 - Can we disable the login/signup buttons until the process is ready? Or possibly make it configurable to hide them? I don't expect we're going to have a public-facing backend server anytime soon, so the login/sign up buttons will never work on the github pages demo


Do we just change the link in the readme to be /demo?

**Comment by dehall** [comment]
Yeah just changing the link sounds fine. Then it's the same as how Camino works with the /demo endpoint.

I'll approve this but not merge it yet - go ahead and update the TODOs

**Review by dehall — APPROVED** [review]

**Pull request merged**
